### ISSUE TITLE
Pipe table parsing deviation

### ISSUE THREAD

**Issue by xxxserxxx** [sivukhin/godjot#6]
godjot's output for pipe tables does not match the reference djot. The issue presents when the table contains multiple headers (which is specifically allowed by the spec).

Here's a small sample to produce the issue:

```djot
| XXX, XXXXX          | XXXXXX          | XXXXXX           |
|---------------------|-----------------|------------------|
| XXXXX XXXX          | XXXX-XX XXXX    | XXXXXXXXX, XX    |
| XXXXX XXXXX (X)     | XX:XXXX         | X:XXXX           |
|                     |                 |                  |
| XXX, XXXXX          | XXXXXX          | XXXXXX           |
|---------------------|-----------------|------------------|
| XXXXX XXXX          | XXXXXXXXX, XX   | XXXXXX, XX       |
| XXXXX XXXXXXX+ (X)  | X:XXXX          | XX:XXXX          |
|                     |                 |                  |
| XXX, XXXXX          | XXXXXX          | XXXXXX           |
|---------------------|-----------------|------------------|
| XXXXX XXXX          | XXXXXX, XX      | XXXX-XX XXXX     |
| XXXXX XXXXX (X)     | X:XXXX          | X:XXXX           |
```

Then to produce the test output:

```bash
$ <sample.dj >godjot.html godjot
$ <sample.dj >djot.html djot
```

and to show the difference, run `diff godjot.html djot.html`:

```diff
13,15c13,20
< <th>XXXXX XXXXX (X)</th>
< <th>XX:XXXX</th>
< <th>X:XXXX</th>
---
> <td>XXXXX XXXXX (X)</td>
> <td>XX:XXXX</td>
> <td>X:XXXX</td>
> </tr>
> <tr>
> <td></td>
> <td></td>
> <td></td>
28,30c33,40
< <th>XXXXX XXXXXXX+ (X)</th>
< <th>X:XXXX</th>
< <th>XX:XXXX</th>
---
> <td>XXXXX XXXXXXX+ (X)</td>
> <td>X:XXXX</td>
> <td>XX:XXXX</td>
> </tr>
> <tr>
> <td></td>
> <td></td>
> <td></td>
```

The difference is significant, by which I mean it's not simply formatting; the parser produces a completely different AST. Or, maybe the HTML renderer is processing the AST incorrectly -- I haven't yet written a sample program to detect which is happening. But I suspect the parser.

I'm running godjot installed with `go install github.com/sivukhin/godjot@latest`, which appears to be v1.0.3:

```
github.com/sivukhin/godjot v1.0.3 h1:ygvKHn+b2vP+lpKYCHT1PCJiSeDJrpJARXmBdWZVCsg=
github.com/sivukhin/godjot v1.0.3/go.mod h1:wA6KdR4Z+XpwdwyViPDLWYYxT72pKjNc6XGA9I025gM=
```

And compiled with `go1.22.4 linux/amd64`

**Comment by sivukhin** [comment]
Hi @xxxserxxx! Thanks for the great sample to reproduce the bug!

The issue were in the pipe separators detection logic. I already fixed it in the PR #7 and published `godjot@v1.0.4` which should resolve your issue. 

Now, `godjot` produce exactly the same output as original `djot` implementation on your example.

**Comment by xxxserxxx** [comment]
Beautiful. Thanks for the quick response!
